### PR TITLE
Feature/tvist1 400 party type refactoring

### PR DESCRIPTION
**Pull request by jekuaitk**
* Added `complainantPartyTypes` and `counterPartyTypes` properties to `Board` entity to make party types configurable
* Displayed party type on list of parties

### CODE REVIEW

File: src/Service/PartyHelper.php [hunk 1]
```diff
@@ -126,4 +126,73 @@ public function handleDeleteParty(CaseEntity $case, Party $party)
...
+    {
+        $complainantRelations = $this->relationRepository
+            ->findBy([
+                'case' => $case,
+                'type' => $this->getComplainantPartyTypesByCase($case),
+                'softDeleted' => false,
+            ])
+        ;
+
+        $complainants = array_map(function ($relation) { return ['party' => $relation->getParty(), 'type' => $relation->getType()]; }, $complainantRelations);
```
Comment (lats1) [inline src/Service/PartyHelper.php:184]: I think something went wrong in the formation of this. The body of the function should be on a new line etc.

File: src/Service/PartyHelper.php [hunk 2]
```diff
@@ -126,4 +126,73 @@ public function handleDeleteParty(CaseEntity $case, Party $party)
...
+
+        $counterPartyRelations = $this->relationRepository
+            ->findBy([
+                'case' => $case,
+                'type' => $this->getCounterPartyTypesByCase($case),
+                'softDeleted' => false,
+            ])
+        ;
+
+        $counterparties = array_map(function ($relation) { return ['party' => $relation->getParty(), 'type' => $relation->getType()]; }, $counterPartyRelations);
```
Comment (lats1) [inline src/Service/PartyHelper.php:194]: I think this should be formatted different. The body of the function should be on a new line etc.

**Review by lats1 — CHANGES_REQUESTED** [review]

File: src/Service/PartyHelper.php (hunk 1 above)
Comment (jekuaitk) [reply src/Service/PartyHelper.php]: Agreed. Increased code readability.

File: src/Service/PartyHelper.php (hunk 2 above)
Comment (jekuaitk) [reply src/Service/PartyHelper.php]: Agreed. Increased code readability.

**Review by lats1 — APPROVED** [review]

**Pull request merged**